pkg/provisioner/ironic/ironic.go

```diff
@@ -1151,6 +1151,42 @@ func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings me
 	return settings, schema, nil
 }
 
+// GetFirmwareComponentsForNode gets all available firmware components for a node and return a list.
+func (p *ironicProvisioner) GetFirmwareComponentsForNode() ([]metal3api.FirmwareComponentStatus, error) {
+	ironicNode, err := p.getNode()
+	if err != nil {
+		return nil, fmt.Errorf("could not get node to retrieve firmware components: %w", err)
+	}
+
+	// Get the components from Ironic via Gophercloud
+	componentList, componentListErr := nodes.ListFirmware(p.client, ironicNode.UUID).Extract()
+
+	if componentListErr != nil {
+		return nil, fmt.Errorf("could not get firmware components for node %s: %w", ironicNode.UUID, componentListErr)
+	}
+	p.log.Info("retrieved firmware components for node", "node", ironicNode.UUID, "size", len(componentList))
```

Comment:
nit: probably remove once you're done testing - we log all components in the reconciler.